cmd/clusterctl/cmd/rollout/undo.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rollout
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+)
+
+// undoOptions is the start of the data required to perform the operation.
+type undoOptions struct {
+	kubeconfig        string
+	kubeconfigContext string
+	resources         []string
+	namespace         string
+	toRevision        int64
+}
+
+var undoOpt = &undoOptions{}
+
+var (
+	undoLong = templates.LongDesc(`
+		Rollback to a previous rollout.`)
+
+	undoExample = templates.Examples(`
+		# Rollback to the previous deployment
+		clusterctl alpha rollout undo machinedeployment/my-md-0
+
+		# Rollback to previous machinedeployment --to-revision=3
+		clusterctl alpha rollout undo machinedeployment/my-md-0 --to-revision=3`)
+)
+
+// NewCmdRolloutUndo returns a Command instance for 'rollout undo' sub command
+func NewCmdRolloutUndo(cfgFile string) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:                   "undo RESOURCE",
+		DisableFlagsInUseLine: true,
+		Short:                 "Undo a cluster-api resource",
```

Comment:
show we be a little bit more specific given that we are supporting only machine deployments?

Reply:
I can add this to the cleanup PR since all commands will need updating.